File: app/src/main/java/com/chikadibia/teesas/fragments/DashboardFragment.java
package com.chikadibia.teesas.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.GridView;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;

import com.chikadibia.teesas.R;
import com.chikadibia.teesas.adapters.GridMenuAdapter;
import com.chikadibia.teesas.model.User;

import java.util.ArrayList;
import java.util.List;

public class DashboardFragment extends Fragment implements  GridView.OnItemClickListener {

    List<String> menu_name;
    List<Integer> menu_icon;
    Context context;
    OnGridItemClickListener onGridItemClickListener;
    User user;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_dashboard, container, false);
        context = getActivity();

        menu_icon = new ArrayList<Integer>();
        menu_name = new ArrayList<String>();


        menu_icon.add(R.drawable.layers);
        menu_name.add(getString(R.string.menu_my_topics));

        menu_icon.add(R.drawable.controller);
        menu_name.add(getString(R.string.menu_take_quiz));

        menu_icon.add(R.drawable.conference);
        menu_name.add(getString(R.string.menu_my_friends));

        menu_icon.add(R.drawable.ttrophy_blue);
        menu_name.add(getString(R.string.menu_results));


        GridView gridview = (GridView)view.findViewById(R.id.gridView);
        GridMenuAdapter adapter = new GridMenuAdapter(getActivity(),menu_name,menu_icon);
        gridview.setAdapter(adapter);
        gridview.setOnItemClickListener(this);

        return view;
    }


    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
//        onGridItemClickListener = (OnGridItemClickListener)context;
    }

    public interface OnGridItemClickListener{
        void onItemClicked(AdapterView<?> adapterView, View view, int i, long l);
    }

    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
        onGridItemClickListener.onItemClicked(adapterView,view, i, l);
    }

}

File: app/src/main/java/com/chikadibia/teesas/fragments/DashboardFragment.kt
package com.chikadibia.teesas.fragments

import android.content.Context
import android.widget.AdapterView.OnItemClickListener
import com.chikadibia.teesas.fragments.DashboardFragment.OnGridItemClickListener
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import com.chikadibia.teesas.R
import android.widget.GridView
import com.chikadibia.teesas.adapters.GridMenuAdapter
import android.widget.AdapterView
import androidx.fragment.app.Fragment
import com.chikadibia.teesas.model.User
import java.util.ArrayList

class DashboardFragment : Fragment(), OnItemClickListener {
    var menu_name: MutableList<String>? = null
    var menu_icon: MutableList<Int>? = null
    var onGridItemClickListener: OnGridItemClickListener? = null
    var user: User? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_dashboard, container, false)
        menu_icon = ArrayList()
        menu_name = ArrayList()
        menu_icon?.add(R.drawable.layers)
        menu_name?.add(getString(R.string.menu_my_topics))
        menu_icon?.add(R.drawable.controller)
        menu_name?.add(getString(R.string.menu_take_quiz))
        menu_icon?.add(R.drawable.conference)
        menu_name?.add(getString(R.string.menu_my_friends))
        menu_icon?.add(R.drawable.ttrophy_blue)
        menu_name?.add(getString(R.string.menu_results))
        val gridview = view.findViewById<View>(R.id.gridView) as GridView
        val adapter = GridMenuAdapter(activity, menu_name, menu_icon)
        gridview.adapter = adapter
        gridview.onItemClickListener = this
        return view
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
    }

    interface OnGridItemClickListener {
        fun onItemClicked(adapterView: AdapterView<*>?, view: View?, i: Int, l: Long)
    }

    override fun onItemClick(adapterView: AdapterView<*>?, view: View, i: Int, l: Long) {
        onGridItemClickListener!!.onItemClicked(adapterView, view, i, l)
    }
}